File: Projects/TransportathonHackathon.API/TransportathonHackathon.Domain/Entities/Company.cs
using Core.Persistence.Entities;
using TransportathonHackathon.Domain.Entities.Identity;

namespace TransportathonHackathon.Domain.Entities
{
    public class Company : Entity
    {
        public Guid AppUserId { get; set; }
        public string CompanyName { get; set; }
        public int DriverCount { get; set; }
        public int CompletedJobsCount { get; set; }

        public virtual AppUser AppUser { get; set; }
        public virtual ICollection<Employee>? Employees { get; set; }
        public virtual ICollection<TransportRequest>? TransportRequests { get; set; }
        public virtual ICollection<Vehicle>? Vehicles { get; set; }
    }
}

File: Projects/TransportathonHackathon.Clients/angular-ui/src/app/models/domain-models/company.ts
import { Employee } from './employee';
import { AppUser } from './appUser';
import { BaseEntity } from '../../core/models/entity';

export interface Company extends BaseEntity {
  appUserId: string;
  companyName: string;
  driverCount: number;
  completedJobsCount: number;
  employees: Employee[] | null;
  appUser: AppUser;
}
